### PR TITLE
fix: scope cluster size estimation to included namespaces

### PR DESCRIPTION
**Pull request by shivansh-gohem**
## Description

When a user runs a namespace-scoped scan (e.g. `--include-namespaces default`), `EstimateClusterSize` previously ignored the namespace filter and queried `.Namespace("")` across all namespaces. On large clusters with thousands of resources, this caused the estimation to return a massive count even for a small, targeted scan, forcing an unnecessary fallback to the slower streaming mode.

This PR updates `EstimateClusterSize` to respect `scanInfo.IncludeNamespaces`:
- When `IncludeNamespaces` is set, each GVR's `LIST` request is scoped to only the specified namespaces.
- When no include filter is set, the existing cluster-wide estimation behaviour is preserved.
- Debug logging is added for failed LIST requests during estimation (previously silently skipped).

The test mock is also updated to capture the namespace parameter, and a new test `TestEstimateClusterSize_RespectsIncludeNamespaces` validates the scoped behaviour.

## Environment

OS: macOS / Linux (any)
Version: master

## Expected behavior

When `--include-namespaces` is specified, `EstimateClusterSize` only counts resources within those namespaces, producing a much tighter estimate that avoids unnecessary streaming fallbacks.

## Additional context

Category: Performance / Correctness

## Summary by CodeRabbit

- **Bug Fixes**
  - Cluster size estimates now correctly honor selected namespace filters.
  - Estimates include resources from every included namespace across supported resource types.
  - Cluster-wide estimates continue to work when no namespace filter is applied.
  - Partial listing failures remain tolerated, while estimates still report an error if no resource lists succeed.

### CODE REVIEW

File: core/pkg/resourcehandler/k8sresources.go [hunk 1]
```diff
@@ -1242,17 +1242,42 @@ func (k8sHandler *K8sResourceHandler) EstimateClusterSize(ctx context.Context, s
 		return 0, fmt.Errorf("kubernetes client not available")
 	}
 
+	namespaces := splitNamespaces(scanInfo.IncludeNamespaces)
```
Comment (matthyx) [inline core/pkg/resourcehandler/k8sresources.go:1245]: Non-blocking: `splitNamespaces` doesn't dedupe. If `--include-namespaces` contains a repeated entry (e.g. `default,default`), the loop below issues one LIST per occurrence and double-counts that namespace's resources, inflating the estimate. Worth a quick `slices.Compact` (after sort) or building a set here, since the whole point of this PR is a tighter estimate — but this is a narrow, self-inflicted edge case (duplicate literal namespace names), not a regression versus current behavior, so I wouldn't hold up the merge for it.

**Review by matthyx — APPROVED** [review]
Verified the fix: `EstimateClusterSize` now scopes its LIST calls to `scanInfo.IncludeNamespaces` via the existing `splitNamespaces` helper, and falls back to the prior cluster-wide behavior when no filter is set. Debug logging on per-GVR LIST failures is a nice addition (previously silent). The new `TestEstimateClusterSize_RespectsIncludeNamespaces` test correctly asserts one query per (GVR, namespace) pair and the resulting total; the mock's namespace-propagation change and the retrofitted signature on existing tests look correct too. CI is green.

Left one non-blocking note on a narrow edge case (duplicate namespace entries aren't deduped, so exact repeats would double-count). Doesn't affect the common path and isn't a regression, so approving.
